docs/section5/Rule5-1-2.md

```diff
@@ -0,0 +1,35 @@
+
+# Envelope - Rule 5-1-2  
+
+**Rule ID:** 5-1-2  
+**Rule Description:** The building shall be modeled so that it does not shade itself.  
+**Rule Assertion:** Baseline RMR Building: surface.does_cast_shade = expected value  
+**Appendix G Section:** Section G3.1-5(b) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:** Table G3.1 Section 5a  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  None  
+
+**Manual Check:** None  
+**Evaluation Context:** Each Data Element  
+**Data Lookup:** None  
+
+## Rule Logic:  
+
+- For each building segment in the Baseline model: ```for building_segment_b in B_RMR.building.building_segments:```  
+
+  - For each thermal_block in building segment: ```thermal_block_b in building_segment_b.thermal_blocks:```  
+  
+    - For each zone in thermal block: ```zone_b in thermal_block_b.zones:```  
+
+      - For each space in zone: ```space_b in zone_b.spaces:```  
+
+        - For each surface in space: ```for surface_b in space_b.surfaces:```  
+
```

Comment:
@xingcx, @JasonGlazer the RMR will be changed to have surfaces at zone level. the RDS will need to be updated after that happens.

Reply:
Agreed. Everything else looks good to me.